src/core/models/answersconfig.js

```diff
@@ -0,0 +1,216 @@
+     * @type {function()}
+     */
+    this.onVerticalSearch = parsedConfig.onVerticalSearch;
+
+    /**
+     * Optional, analytics callback after a universal search, see onUniversalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onUniversalSearch = parsedConfig.onUniversalSearch;
+
+    /**
+     * Optional, opt-out of automatic css variable resolution on init for legacy browsers
+     * @type {boolean}
+     */
+    this.disableCssVariablesPonyfill = parsedConfig.disableCssVariablesPonyfill;
+
+    /**
+     * Optional, the analytics key describing the Answers integration type. Accepts 'STANDARD' or 'OVERLAY', defaults to 'STANDARD'
+     * @type {string}
+     */
+    this.querySource = parsedConfig.querySource;
+
+    this.mock = parsedConfig.mock;
+    this.analyticsOptions = parsedConfig.analyticsOptions;
+    this.fieldFormatters = parsedConfig.fieldFormatters;
+
+    this.validate();
+    Object.freeze(this);
+  }
+
+  /**
+   * Validates the Answers config object to ensure things like api key and experience key are
+   * properly set.
+   */
+  validate () {
+    if (typeof this._apiKey !== 'string') {
+      throw new Error('Missing required `apiKey`. Type must be {string}');
+    }
+
+    if (typeof this.onStateChange !== 'function') {
+      throw new Error('onStateChange must be a function. Current type is: ' + typeof this.onStateChange);
+    }
+
+    if (typeof this.experienceKey !== 'string') {
+      throw new Error('Missing required `experienceKey`. Type must be {string}');
+    }
+
+    if (typeof this.onVerticalSearch !== 'function') {
+      throw new Error('onVerticalSearch must be a function. Current type is: ' + typeof this.onVerticalSearch);
+    }
+
+    if (typeof this.onUniversalSearch !== 'function') {
+      throw new Error('onUniversalSearch must be a function. Current type is: ' + typeof this.onUniversalSearch);
+    }
+  }
+
+  /**
+   * The Yext Answers API Key. If the apiKey contains the sandbox prefix, then it is removed leaving only the API key.
+   * @returns {string}
+   */
+  get apiKey () {
+    return this._apiKey.replace(sandboxPrefix, '');
+  }
+
+  /**
+   * The internal Yext Environment.
+   *  Defaults to Production
+   *  If the apiKey contains the sandbox prefix, then the environment returned is Sandbox.
+   * @returns {string}
+   */
+  get environment () {
+    if (this._apiKey.includes(sandboxPrefix)) {
+      return SANDBOX;
+    } else {
+      return this._environment;
+    }
+  }
+
+  /**
+   * Parses the config provided by the user. In the parsed config, any options not supplied by the
+   * user are given default values.
+   * @param {Object} config The user supplied config.
+   */
+  parseConfig (config) {
+    const parsedConfig = Object.assign({}, DEFAULTS, config);
+    let sessionTrackingEnabled = true;
+
+    if (typeof config.sessionTrackingEnabled === 'boolean') {
+      sessionTrackingEnabled = config.sessionTrackingEnabled;
+    }
```

Comment:
this feels weird, maybe we should be catching this in the validation phase

Reply:
i agree, this makes more sense in the validation phase